lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -791,6 +791,26 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value other = convertScalarToDtype(b, loc, operands[1], dtype);
     return b.create<arith::DivFOp>(loc, self, other);
   }
+  if (auto remScalar = dyn_cast<AtenRemainderScalarOp>(op)) {
+    MLIRContext *context = op->getContext();
+    Type newResultType = converter->convertType(remScalar.getType())
+                             .cast<RankedTensorType>()
+                             .getElementType();
+
+    Value self = convertScalarToDtype(b, loc, payloadArgs[0], newResultType);
+    Value other = convertScalarToDtype(b, loc, operands[1], newResultType);
+    Value result; 
+
+    if (newResultType.isa<mlir::FloatType>()){
+      result = b.create<arith::RemFOp>(loc, self, other);
+    }
+    else{
+      result = b.create<arith::RemSIOp>(loc, self, other);
+    }
+
+    result = convertScalarToDtype(b, loc, result, newResultType);
```

Comment:
Is this extra cast needed?

Reply:
I don't think its needed i forgot to remove it will do that now.